repo-agent/review-ui/ui/src/UpdateRepo.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+import yaml from 'js-yaml';
+
+function UpdateRepo({ repo, onCancel, onRepoUpdated }) {
+    const [yamlContent, setYamlContent] = useState('');
+    const [originalYamlContent, setOriginalYamlContent] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState('');
+
+    useEffect(() => {
+        setIsLoading(true);
+        fetch(`/api/repos/${repo.name}/yaml`)
+            .then(res => {
+                if (!res.ok) throw new Error("Failed to fetch repo YAML");
+                return res.json();
+            })
+            .then(data => {
+                if (data.yaml) {
+                    setYamlContent(data.yaml);
+                    setOriginalYamlContent(data.yaml); // Store original YAML content
+                }
+                setIsLoading(false);
+            })
+            .catch(err => {
+                console.error(err);
```

Comment:
The error handling for this API call only logs to the console. If the request fails, the user will see a blank editor with no indication that something went wrong. You should implement user-facing error feedback, such as displaying an alert or an error message on the screen.